src/components/home/LatestTransactions.tsx

```diff
@@ -1,21 +1,35 @@
 import { useTranslation } from 'react-i18next';
-import { EmptyConnectionsIcon } from '@/shared/components';
+import { ConfirmedTransactionData } from '@ardenthq/sdk/distribution/esm/confirmed-transaction.dto.contract';
+import { useEffect, useState } from 'react';
+import { NoTransactions, TransactionsList } from './LatestTransactions.blocks';
+import { usePrimaryWallet } from '@/lib/hooks/usePrimaryWallet';
 
-const NoTransactions = () => {
+export const LatestTransactions = () => {
     const { t } = useTranslation();
+    const primaryWallet = usePrimaryWallet();
 
-    return (
-        <div className='mt-12 flex flex-col items-center justify-center gap-6'>
-            <EmptyConnectionsIcon />
-            <div className='max-w-40 text-center text-base font-normal leading-tight dark:text-white'>
-                {t('PAGES.HOME.NO_TRANSACTIONS')}
-            </div>
-        </div>
-    );
-};
+    const [transactions, setTransactions] = useState<ConfirmedTransactionData[]>([]);
 
-export const LatestTransactions = () => {
-    const { t } = useTranslation();
+    const fetchTransactions = async () => {
+        try {
+            const response = await primaryWallet?.client().transactions({
```

Comment:
instead of directly using the client like this, you want the psdk to do the fetching for you to return proper transaction objects (with additional methods becoming available as opposed to these bare transaction objects you retrieve now).

For this you want to make use of `const response = await primaryWallet?.transactionIndex().all({ limit: 10 });` instead of the client. This will return an `ExtendedConfirmedTransactionDataCollection` instead and you can loop over those to get the data you need for the overview.

The client should only really be used when you need to fetch data from the API directly that you cannot get otherwise in the PSDK